packages/labs/src/frameworks/react/create-component.ts

```diff
@@ -0,0 +1,216 @@
+  // and not handled directly by React.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const elementPropsMap: {[index: string]: any} = {...(events ?? {})};
+  for (const p in elementClass.prototype) {
+    if (!(p in HTMLElement.prototype)) {
+      elementPropsMap[p] = true;
+    }
+  }
+
+  class ReactComponent extends Component<ComponentProps> {
+    private _element: I | null = null;
+
+    private _elementProps!: typeof elementPropsMap;
+
+    private _ref = (element: I | null) => {
+      this._element = element;
+      // Fulfill any ref we receive in our props.
+      // The `ref` prop is special and cannot be accessed without an error
+      // so use `__forwardedRef`.
+      const forwardedRef = this.props?.__forwardedRef;
+      if (forwardedRef) {
+        // Note, there are 2 kinds of refs and we manually fulfill them here.
+        // There is no built in React API for this.
+        if (typeof forwardedRef === 'function') {
+          (forwardedRef as (e: I | null) => void)(element);
+        } else {
+          // This has to be `any` because `current` is readonly.
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          (forwardedRef as any).current = element;
+        }
+      }
+    };
+
+    private _updateElement(oldProps?: ComponentProps) {
+      if (this._element === null) {
+        return;
+      }
+      // Set element properties to the values in `this.props`
+      for (const prop in this._elementProps) {
+        setProperty(
+          this._element,
+          prop,
+          this.props[prop as keyof ComponentProps],
+          oldProps ? oldProps[prop as keyof ComponentProps] : undefined,
+          events
+        );
+      }
+      // Note, the spirit of React might be to "unset" any old values that
+      // are no longer included; however, there's no reasonable value to set
+      // them to so we just leave the previous state as is.
+    }
+
+    /**
+     * Updates element properties correctly setting properties
+     * on mount.
+     */
+    componentDidMount() {
+      this._updateElement();
+    }
+
+    /**
+     * Updates element properties correctly setting properties
+     * on every update. Note, this does not include mount.
+     */
+    componentDidUpdate(old: ComponentProps) {
+      this._updateElement(old);
+    }
+
+    /**
+     * Renders the custom element with a `ref` prop which allows this
+     * component to reference the custom element.
+     *
+     * Standard attributes are passed to React and element properties and events
+     * are updated in componentDidMount/componentDidUpdate.
+     *
+     */
+    render() {
+      // Filters class properties out and passes the remaining
+      // attributes to React. This allows attributes to use framework rules
+      // for setting attributes and render correctly under SSR.
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const props: any = {ref: this._ref};
```

Comment:
See if `props: object` or `props: {}` lets you remove the `any`.

Reply:
Isn't it `Partial<ComponentProps>`? And then can get rid of the `p as keyof ComponentProps` below.

Reply:
The `any` helps deal with the finicky React type. Using `ComponentProps` is problematic with the `ref` type. Using object or an index type doesn't work either.